machine/v1/types_alibabaprovider.go

```diff
@@ -0,0 +1,340 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AlibabaDiskPerformanceLevel enum attribute to describe a disk's performance level
+type AlibabaDiskPerformanceLevel string
+
+// AlibabaDiskCatagory enum attribute to deescribe a disk's category
+type AlibabaDiskCategory string
+
+// AlibabaDiskEncryptionMode enum attribute to describe whether to enable or disable disk encryption
+type AlibabaDiskEncryptionMode string
+
+// AlibabaDiskPreservationPolicy enum attribute to describe whether to preserve or delete a disk upon instance removal
+type AlibabaDiskPreservationPolicy string
+
+const (
+	// DeleteWithInstance enum property to delete disk with instance deletion
+	DeleteWithInstance AlibabaDiskPreservationPolicy = "DeleteWithInstance"
+	// PreserveDisk enum property to determine disk preservation with instance deletion
+	PreserveDisk AlibabaDiskPreservationPolicy = "PreserveDisk"
+
+	// AlibabaDiskEncryptionEnabled enum property to enable disk encryption
+	AlibabaDiskEncryptionEnabled AlibabaDiskEncryptionMode = "encrypted"
+	// AlibabaDiskEncryptionDisabled enum property to disable disk encryption
+	AlibabaDiskEncryptionDisabled AlibabaDiskEncryptionMode = "disabled"
+
+	// AlibabaDiskPerformanceLevel0 enum property to set the level at PL0
+	PL0 AlibabaDiskPerformanceLevel = "PL0"
+	// AlibabaDiskPerformanceLevel1 enum property to set the level at PL1
+	PL1 AlibabaDiskPerformanceLevel = "PL1"
+	// AlibabaDiskPerformanceLevel2 enum property to set the level at PL2
+	PL2 AlibabaDiskPerformanceLevel = "PL2"
+	// AlibabaDiskPerformanceLevel3 enum property to set the level at PL3
+	PL3 AlibabaDiskPerformanceLevel = "PL3"
+
+	// AlibabaDiskCategoryUltraDisk enum proprty to set the category of disk to ultra disk
+	AlibabaDiskCatagoryUltraDisk AlibabaDiskCategory = "cloud_efficiency"
+	// AlibabaDiskCategorySSD enum proprty to set the category of disk to standard SSD
+	AlibabaDiskCatagorySSD AlibabaDiskCategory = "cloud_ssd"
+	// AlibabaDiskCategoryESSD enum proprty to set the category of disk to ESSD
+	AlibabaDiskCatagoryESSD AlibabaDiskCategory = "cloud_essd"
+	// AlibabaDiskCategoryBasic enum proprty to set the category of disk to basic
+	AlibabaDiskCatagoryBasic AlibabaDiskCategory = "cloud"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlibabaCloudMachineProviderConfig is the Schema for the alibabacloudmachineproviderconfig API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type AlibabaCloudMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
```

Comment:
something i am confused/concerned about with our migration of provider specs to the openshift/api is that it seems like these types will now have different api groups than they previously had. eg, aws in 4.9 would have been `awsproviderconfig.openshift.io`, but will now be `machine.openshift.io`. i presume that this alibaba provider config will have the `machine.openshift.io` group?

@JoelSpeed @alexander-demichev , i don't remember this coming up during our discussions of the api move, but i also admit this point is highly nuanced so i might have missed the detail back then. is there perhaps a kubebuilder annotation that we can add to adjust the group for these things?

given that we have these embedded api types documented at specific groups in [our documentation](https://docs.openshift.com/container-platform/4.9/machine_management/creating_machinesets/creating-machineset-aws.html#machineset-yaml-aws_creating-machineset-aws) already, i'm really curious how much of an issue this is. it seems like a big deal, but i might be missing something about our api types. i looked back at the discussion in #1005 when the provider specs were added but i didn't see any discussion there. what do you think?

Reply:
we discussed this issue during our team standup today. the team feels that the change of group for these types is extremely low risk. these objects do not persist in the kuberenetes api server beyond being data in raw extensions, and as such are not being used as kubernetes objects in the manner we would expect of an object like a Machine.